bcgov/cas-registration
# Test/90 model tests

Addresses #90 [Zenhub](https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/90). Adds Automated testing of the Django models. And further suggested tests would be appreciated!

## Changes 🚧

- Added tests for all existing models 

## To test 🔬

1. Navigate to the `~/bc_obps` directory, and ensure you are in the Poetry shell. Start Postgres with `make start_pg`. Follow the [developer environment setup](https://github.com/bcgov/cas-registration/blob/develop/docs/developer-environment-setup.md#backend-environment-setup) if you need to create the DB locally.
2. Run `python manage.py test`. It should find and run 106 tests. 

### Notes 📝

These tests might not be complete. Following the paradigms outlined by the [Django Project](https://docs.djangoproject.com/en/4.2/topics/testing/overview/) and  [Mozilla](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Testing), I've included tests for code defined by us rather than Django. These tests will need to be extended if methods are ever added to the model classes.
